x-pack/plugins/alerts/server/lib/retry_if_conflicts.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// This module provides a helper to perform retries on a function if the
+// function ends up throwing a SavedObject 409 conflict.  This can happen
+// when alert SO's are updated in the background, and will avoid having to
+// have the caller make explicit conflict checks, where the conflict was
+// caused by a background update.
+
+import { random } from 'lodash';
+import { Logger, SavedObjectsErrorHelpers } from '../../../../../src/core/server';
+
+type RetryableForConflicts<T> = () => Promise<T>;
+
+// number of times to retry when conflicts occur
+export const RetryForConflictsAttempts = 5;
```

Comment:
This feels high 🤔 

I get why we're allowing it a second attempt, but 5 times feels like we might be addressing a symptom instead of the root cause. Any thoughts?

Reply:
Ya, I was just thinking today, seems likely you'd never have to retry more than once.  And 5 does seem pretty high.  If you can't update within 5 retries, probably something is very wrong with your system :-).  

OTOH, if you do get to this horrible place, these APIs could provide some "breathing room" for the system to fix itself - that's a bit of a stretch I'd say.

What kinda number you thinking?  I could see going as low as 2, I think 3 feels a little better to me.

Reply:
After conversation with Gidi, changed the retries to `1` - we'll find out from the SIEM tests once we add the alert execution status update, if this is good enough.